torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -29,9 +29,15 @@ namespace rpc {
 
 using namespace torch::distributed::autograd;
 
-Message RequestCallbackImpl::processRpc(
+std::shared_ptr<FutureMessage> RequestCallbackImpl::processRpc(
     RpcCommandBase& rpc,
-    MessageType messageType) const {
+    MessageType messageType,
+    int64_t messageId) const {
+  auto wrap = [messageId](Message m) {
+    m.setId(messageId);
+    return std::make_shared<FutureMessage>(std::move(m));
```

Comment:
sorry, I'm a little confused about how this is async. Would you please explain more? IIUC, when the futureMessage is created here, message has already been processed and is completed, this looks like it is the same as current sync behavior. I thought we should immediately return a futureMessage in some way and mark the message later on/asynchronously?

Reply:
Right, the parts exiting with wrap() and returning a completed FutureMessage are definitely not async :)

In a previous iteration, I had stacked in dist autograd changes which did return an async FutureMessage, though seemed sensible to separate out to reduce merge risk.  (Still trying to gauge this team's culture with respect to spanning changes)

Reply:
Actually, right after typing this, I just got tests to pass with my async BACKWARD_AUTOGRAD_REQ impl (I had returned an incomplete future in a base case that I should have returned a completed one).

So, I updated this change to include async for BACKWARD_AUTOGRAD_REQ, which should be a win here,